src/scores/probability/crps_impl.py

```diff
@@ -1094,3 +1096,101 @@ def _vfunc(x, threshold=threshold):
         weights=weights,
     )
     return result
+
+
+def interval_tw_crps_for_ensemble(
+    fcst: XarrayLike,
+    obs: XarrayLike,
+    ensemble_member_dim: str,
+    lower_threshold: Union[xr.DataArray, float],
+    upper_threshold: Union[xr.DataArray, float],
+    *,  # Force keywords arguments to be keyword-only
+    method: Literal["ecdf", "fair"] = "ecdf",
+    reduce_dims: Optional[Sequence[str]] = None,
+    preserve_dims: Optional[Sequence[str]] = None,
+    weights: Optional[XarrayLike] = None,
+) -> XarrayLike:
+    """
+    Calculates the threshold weighted continuous ranked probability score (twCRPS)
+    weighted for an interval across the distribution from ensemble input.
+
+    A threshold weight of 1 is assigned for values within the interval and a threshold weight of 0 otherwise.
```

Comment:
Do people always want a step function, or would some kind of ramping be useful?

Reply:
Potentially people may want ramping. For now, I suggest people supply their own custom chaining function to `tw_crps_for_ensemble` if they want to do fancier things like this.

I think that it's a nice idea to add that capability in the future.

Reply:
Added an issue for this here https://github.com/nci/scores/issues/691